plugins/services/src/js/containers/services/ServicesTable.js

```diff
@@ -651,4 +657,28 @@ ServicesTable.propTypes = {
   services: PropTypes.array
 };
 
-module.exports = ServicesTable;
+// TODO: move into it's own file
+function withMasterRegionName(Component) {
+  const master$ = MesosStateStore.getMasterRequestStream();
+  const masterRegionName$ = master$.pipe(
+    map(masterRequestString => {
+      const masterRequest = JSON.parse(masterRequestString);
```

Comment:
Not important now, would there be a way to share this parsed version of the string?  I imagine most examples that follow this pattern would want to get the object to find the specific key(s) for that component?

Reply:
Yeah, I'm planning on moving the data parsing up to the nodes store. The only thing I'm a little worried about is that we now have this piece of data selection in the component. I'm thinking about time boxing a data layer effort here to an hour to have this nice encapsulated and still a good performance tradeoff